plugins/woocommerce/includes/class-woocommerce.php

```diff
@@ -546,7 +548,7 @@ public function includes() {
 			include_once WC_ABSPATH . 'includes/class-wc-cli.php';
 		}
 
-		if ( $this->is_request( 'admin' ) ) {
+		if ( $this->is_request( 'admin' ) || $this->is_request( 'rest' ) ) {
```

Comment:
Note that this includes everything in `class-wc-admin.php` in REST requests.  Luckily, all logic is housed within hooks such as `admin_init` that will not be triggered on REST requests unless specifically added.

But this constructor should be audited to make sure there are no extra functions that will accidentally get triggered during REST requests.

Reply:
Do you have something in mind for this audit? As it stands we can't be 100% that the logic within in `class-wc-admin.php` will remain behind non-rest-compatible hooks, although foundational changes seem unlikely.

Reply:
I was mostly thinking just a manually look-through would suffice as I don't see anything that would currently be exposed currently.

>we can't be 100% that the logic within in class-wc-admin.php will remain behind non-rest-compatible hooks

Good point.  Going forward, it might be good to either:

a. Add an additional check and not load hooks if in a REST request.  If nothing else, this will bring awareness to the fact that these hooks may be exposed during REST requests.

b. Refactor the tracking logic out of the "admin" classes.

Reply:
> a. Add an additional check and not load hooks if in a REST request. If nothing else, this will bring awareness to the fact that these hooks may be exposed during REST requests.

I like the idea of at least adding some visibility, to your point. 

> b. Refactor the tracking logic out of the "admin" classes.

This feels like the "cleaner" approach. Do you have a grasp of the downsides here? Without looking into it further I'd expect  a bit more chance of side-effects and more time to implement.

Reply:
Total drive-by comment here (prompted by Josh's Slack comment), so I may be missing context/understanding... why not just call `WC_Site_Tracking::init()` in `load_rest_api()`?

Reply:
I like @mattsherman idea, although not overly opposed to your current solution.
I do have one follow up question, are we able to tell if a REST request is coming from `admin` or not? Maybe this is worth checking? Maybe looking at the Referer?
```
$referer = wp_get_referer();
$admin_rest_request = $referer && 0 === strpos( $referer, wc_admin_url() );
```

Reply:
If we were to determine if the REST request was coming from `admin`, what would we do with that information? I think we'd want to log Tracks events in REST API calls regardless of where the request was from. Otherwise, we potentially have "hidden" execution of code that might confuse our analyses. Perhaps it would be useful to log whether it was from `admin` or not though!

Reply:
Yeah that makes sense, I guess it would depend on what our priority is around this. Given this condition in particular includes the `class-wc-admin.php` file, I figured we are better of double checking if the request is coming from `admin` as well.

Reply:
>Total drive-by comment here (prompted by Josh's Slack comment), so I may be missing context/understanding... why not just call WC_Site_Tracking::init() in load_rest_api()?

@mattsherman I _think_ that some of the tracking logic is a little too tightly coupled with the `WC_Admin` class right now.  So this is also my preferred solution, but probably leans in a little towards the refactor I mentioned above.

Happy to do that in this PR if we agree that's the best approach.

> I think we'd want to log Tracks events in REST API calls regardless of where the request was from.

➕ 1️⃣

Reply:
> I think that some of the tracking logic is a little too tightly coupled with the WC_Admin class right now.

@joshuatf Do you have any examples of this? I'm not familiar with this, but I don't see why calling WC_Site_Tracking::init() in load_rest_api() wouldn't work.

Reply:
@mattsherman One of the most obvious offenders is the tracks class, events, and other various functionality being included as part of the `WC_Admin` class:

https://github.com/woocommerce/woocommerce/blob/fix/37659/plugins/woocommerce/includes/admin/class-wc-admin.php#L71-L75

This probably won't be painfully hard to refactor, but one area that's not so obvious is all the included helper/util files.  Global functions may be used inside tracks functions that won't be available without these includes.  I'm worried the only way to surface and test these would be to go through each and every tracks event, which may be error prone and time consuming.

I can give this a quick shot and see if anything immediately pops out as problematic.